### PR TITLE
modmod: Support images in exercise descriptions

### LINKED ISSUE #118: modmod: Support images in exercise descriptions
What do you think about adding support for images / other static files in exercise descriptions?
Or is this already supported somehow?

Usecase: I am working on an exercise for wasm and I would like to show an image of the result in the exercise description (of course, it also would be cool to embed some wasm animation).

### PR DESCRIPTION
**Pull request by g-braeunlich**
Closes #118

One problem to be discussed:
The current approach does not handle image conflicts (two exercises having images with the same name).
With the current implementation, the images of an exercise with a higher index just would overwrite images from exercises with a lower index.

### CODE REVIEW

**Review by hdoordt — CHANGES_REQUESTED** [review]
Thanks again @g-braeunlich, this looks great in general!

I think we can mitigate the issue with multiple exercises referring to images with the same name, by requiring the markdown to prefix the image name with a `#[modmod:exercise_images]`, like the [others defined here](https://github.com/tweedegolf/teach-rs/blob/63f08316ceb02913ad6b424f91f2b3064624a99c/modmod/src/book.rs#L96-L106) which resolves to a directory that is unique to the exercise within the `images` directory, and place the images in said directory.

This does require a little bit more plumbing

**Comment by g-braeunlich** [comment]
Sorry for the late response. But now I think I have implemented your suggestion. It also will create a `images/{chapter_i}/{section_i}/{subsection_i}` folder hierarchy in the output `src` folder.
And I also implemented a check which will print warnings for non existing images / unreferenced images.

**Comment by g-braeunlich** [comment]
Would you rather have exceptions instead of warnings?

**Comment by hdoordt** [comment]
@g-braeunlich sorry for the delay. I think exceptions would be nicer, as this is something you'd want to be rather 'in your face' when setting up a module. Very nice otherwise!

File: modmod/src/book.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ pub struct Book<'track> {
     pub chapters: Vec<Chapter<'track>>,
 }
 
+const IMAGE_PLACEHOLDER: &str = "#[modmod:images]";
```
Comment (hdoordt) [inline modmod/src/book.rs:33]: This is nice. Could you make the other placeholders to consts too?

**Review by hdoordt — COMMENTED** [review]
Very nice so far, thanks already!

**Comment by hdoordt** [comment]
With #116, I refactored modmod quite heavily, which will make it non-trivial to merge this PR. I will however rebase this branch and resolve the conflicts once you're happy with your contribution

File: modmod/src/book.rs (hunk 1 above)
Comment (g-braeunlich) [reply modmod/src/book.rs:40]: Should be done now

**Comment by hdoordt** [comment]
@g-braeunlich Thanks!

**Pull request merged**